api/views/student.py

```diff
@@ -109,6 +109,42 @@ def missed_class(self, request: Request, personal_info_id: int) -> Response:  #
         )
         return Response(status=status.HTTP_204_NO_CONTENT)
 
+    @extend_schema(
+        responses={
+            status.HTTP_204_NO_CONTENT: OpenApiResponse(description="Action is taken"),
+            status.HTTP_409_CONFLICT: OpenApiResponse(
+                response=ValidationErrorSerializer,
+                description="Only students with no groups can be processed",
+            ),
+        },
+    )
+    @action(detail=True, methods=["post"])
+    def finished_and_left(
+        self, request: Request, personal_info_id: int  # noqa: ARG002
+    ) -> Response:
+        student = self.get_object()
+        if student.has_groups:
+            raise ConnectionError("Only students with no groups can be processed")
```

Comment:
just checking: is `ConnectionError` really an appropriate exception here?

Reply:
Good you noticed, it is wrong